kolibri/plugins/document_pdf_render/assets/src/vue/index.vue

```diff
@@ -51,11 +51,17 @@
         }
       },
     },
-
     ready() {
       PDFobject.embed(this.defaultFile.storage_url, this.$els.pdfcontainer);
+      this.$emit('startTracking', this.Kolibri);
+      const self = this;
+      setTimeout(() => {
+        self.$emit('progressUpdate', this.Kolibri, 1);
+      }, 15000);
+    },
+    beforeDestroy() {
+      this.$emit('stopTracking', this.Kolibri);
```

Comment:
Perhaps cancel that `setTimeout` in case the user exits the content before 15 secs has elapsed.